### ISSUE TITLE
Speed Hacks

### ISSUE THREAD

**Issue by kumavis** [ConsenSys-archive/ganache-cli-archive#2]
Here are some sneaky things you can do to get some speedups
- [ ] use [the fake stateTrie](https://github.com/kumavis/fake-merkle-patricia-tree) if you don't care about state roots
  ( you can still checkpoint and revert )
- [x] skip proper tx signing by overriding the `from:` field
- [x] skip tx signing for eth_call's

``` js
transaction.from = new Buffer(ethUtil.stripHexPrefix(txParams.from), 'hex')
```

**Comment by tcoulter** [comment]
Added to v0.2.0 milestone (next release)

**Comment by tcoulter** [comment]
Need to double check - do we care about state roots? 

Also, skipping transaction signing still needs to happen for eth_calls. Added to ticket.

**Comment by kumavis** [comment]
i think the testrpc api has a thing for checkpointing that relies on the state roots
the fakeTrie has support for checkpointing but only in the checkpoint/commit/revert sense, you cant just set a previous stateRoot ( bc all stateRoots are `0x00000...0000` )

otherwise you dont need the stateroots if you are not importing blocks and comparing the result... i think... maybe transactionReceipt includes them tho?

**Comment by tcoulter** [comment]
Closing this one. I think this is all implemented now, both in the provider engine and the TestRPC. The fake statetrie could still be investigated, but I'd like to do that in another ticket if there's still interest.

**Comment by kumavis** [comment]
I think you'll find a big speed up from the fake stateTrie, but as it is you are prolly not blocked on perf
